### PR TITLE
Feature/animated-sleeping

### PR DESCRIPTION
**Pull request by exincore**
# Animated Sleeping

##### What?

- The cat has a sleeping animation now
- Varying number of sprites can be used in animations without needing to modify code

##### Why?

- It is adorable
- They need a nice nap after running
- Greater support for different animation sets such as #29 

##### How?

- `panelMenuButton.js` looks for and loads sprites into arrays based on their name like `active-0` or `idle-3` etc.
- `prefs.js` et al. has a toggle button for reverting back to non-animated sleeping
- Brand new sleeping sprites based on the original `my-sleeping-symbolic.svg` (contains cute snoozing)

my-sleeping-3-symbolic

### CODE REVIEW

**Comment by win0err** [comment]
Hello, @EXINCORE!

Thanks for the PR! I need some time to review the PR.

File: src/panelMenuButton.js [hunk 1]
```diff
@@ -8,6 +8,8 @@ const { Timer } = Extension.imports.timer;
 const { Cpu } = Extension.imports.cpu;
 const { IconProvider } = Extension.imports.iconProvider;
 
+const { states } = Extension.imports.iconProvider;
```
Comment (EXINCORE) [inline src/panelMenuButton.js:11]: gnome-shell warns this is against ES6 standard

Comment (EXINCORE) [inline src/panelMenuButton.js:11]: gnome-shell warns this is against ES6 standard

File: src/panelMenuButton.js [hunk 2]
```diff
@@ -116,10 +123,13 @@ var PanelMenuButton = GObject.registerClass(
...
-                            const isRunningSpriteShown = this.cpu.utilization > this.sleepingThreshold;
-                            this.ui.get('icon').set_gicon(
-                                isRunningSpriteShown ? this.iconProvider.nextSprite : this.iconProvider.sleeping,
-                            );
+                            if (this.cpu.utilization > this.sleepingThreshold) {
+                                this.iconProvider.animationState = states.RUNNING;
+                            } else {
+                                this.iconProvider.animationState = this.animatedSleeping
+                                    ? states.SLEEPING : states.SLEEPING_STATIC;
+                            }
```
Comment (EXINCORE) [inline src/panelMenuButton.js:131]: `iconProvider` might need to handle this state management code in order to avoid using `'states'` outside its module.

Comment (EXINCORE) [inline src/panelMenuButton.js:131]: `iconProvider` might need to handle this state management code in order to avoid using `'states'` outside its module.

**Comment by win0err** [comment]
Please, fix conflicts and I will merge the PR

**Comment by win0err** [comment]
Closed by mistake

**Comment by exincore** [comment]
Goodness, this will take a while to rebase

**Comment by exincore** [comment]
@win0err I will clean up the idle sprites, but other than that, this feature is ready for re-review.

File: po/ru.po [hunk 3]
```diff
@@ -47,47 +47,51 @@ msgid "Idle threshold"
 msgstr "Порог бездействия"
 
 #: src/resources/ui/preferences.ui:34
+msgid "Idle animation"
+msgstr ""
```
Comment (exincore) [inline po/ru.po:51]: One new translation is needed

File: po/messages.pot [hunk 4]
```diff
@@ -46,45 +46,49 @@ msgid "Idle threshold"
 msgstr ""
 
 #: src/resources/ui/preferences.ui:34
+msgid "Idle animation"
```
Comment (win0err) [inline po/messages.pot:49]: Wouldn't it be better to call it `"Animated idle"`?

Comment (exincore) [reply po/messages.pot:49]: I decided on `Idle animation` due to the presence of `Idle threshold`, however I will go ahead and rename everything.

**Comment by win0err** [comment]
LGTM. Great work, thank you!

**Pull request merged**